viewing_party/party.py

```diff
@@ -1,23 +1,139 @@
 # ------------- WAVE 1 --------------------
 
 def create_movie(title, genre, rating):
-    pass
+    
+    if title and genre and rating:
+        return {
+        "title": title,
+        "genre": genre, 
+        "rating": rating
+        }
+    else:
+        return None
+        
+def add_to_watched(user_data, movie):
+    user_data["watched"].append(movie)
+    return user_data 
+
+def add_to_watchlist(user_data, movie):
+    user_data["watchlist"].append(movie)
+    return user_data
+
+def watch_movie(user_data, title):
+    # watchlist = user_data["watchlist"]
+    # watched = user_data["watched"]
+
+    for movie in user_data["watchlist"]:
+        if movie["title"] == title:
+            user_data["watched"].append(movie)
+            user_data["watchlist"].remove(movie)
+    return user_data
+
+
+
 
 # -----------------------------------------
 # ------------- WAVE 2 --------------------
 # -----------------------------------------
+def get_watched_avg_rating(user_data):
+    list_of_ratings = []
+    for movie in user_data["watched"]:
+        list_of_ratings.append(movie["rating"])
+    if len(user_data["watched"]) == 0:
+        avg_rating = 0
+    else:
+        avg_rating = sum(list_of_ratings)/len(list_of_ratings)
+    return avg_rating
+
+def get_most_watched_genre(user_data):
+    genre_list = []
+    for movie in user_data["watched"]:
+        genre_list.append(movie["genre"])
+
+    if user_data["watched"] == []:
+        return None
+    else:
+        counter = 0
+        for genre in genre_list:
+            frequency = genre_list.count(genre)

```

Comment:
The `list` `count` method must iterate over the entire list in order to count the number of occurrences. So line 57 iterates over the `genre_list` contents, and then for each value, finds the count by iterating over `genre_list` again, a quadratic O(n^2) time complexity. Rather than using `count` we could iterate through the genre data, and pack it into a dictionary where the genere name is the key and the value is the number of times we've seen the genre (starting at 1 for the first time we encounter the genre--the genre wouldn't yet appear in the dict--and adding one eto the value each time thereafter). Then we could either keep track of the highest genre we've seen so far, or do one more iteration at the end to find the genre with the maximum count.

This would have a linear O(n) time complexity rather than the O(n^2) we have here.